charmcraft/utils.py

```diff
@@ -136,3 +140,52 @@ def useful_filepath(filepath):
     if not filepath.is_file():
         raise CommandError("{!r} is not a file.".format(str(filepath)))
     return filepath
+
+
+def get_os_platform(filepath=pathlib.Path("/etc/os-release")):
+    """Determine a system/release combo for an OS using /etc/os-release if available."""
+    system = platform.system()
+    release = platform.release()
+    machine = platform.machine()
+
+    if system == "Linux":
+        os_release = {}
+        try:
+            with filepath.open("r", encoding='utf-8') as f:
+                for line in f:
+                    if "=" in line:
+                        key, value = line.rstrip().split("=", 1)
+                        os_release[key] = value.strip('"')
+        except FileNotFoundError:
+            logger.debug("Unable to locate 'os-release' file, using default values")
+        finally:
+            system = os_release.get("NAME", system)
+            release = os_release.get("VERSION_ID", release)
+
+    return OSPlatform(system=system, release=release, machine=machine)
+
+
+def create_manifest(basedir, started_at):
+    """Save context information for the charm execution.
+
+    Mostly to be used by builders.
+    """
+    os_platform = get_os_platform()
+
+    # XXX Facundo 2021-03-29: the architectures list will be provided by the caller when
+    # we integrate lifecycle lib in future branches
+    architectures = [os_platform.machine]
+
+    content = {
+        'charmcraft-version': __version__,
+        'charmcraft-started-at': started_at.isoformat() + "Z",
```

Comment:
It seems that we're converting from ISO 8601 to RFC3339 date formats here, is the original `started_at` time UTC+0 so we can safely say it's Zulu time?